File: Chess/src/main/java/side_server/ListenConnectionRequestThread.java
package side_server;

import java.io.IOException;
import java.net.Socket;


public class ListenConnectionRequestThread extends Thread {
    public Server server;

    public ListenConnectionRequestThread(Server server) {
        this.server = server;
    }

    @Override
    public void run() {
        while (!this.server.socket.isClosed()) {
            try {
                Socket nSocket = this.server.socket.accept();
                SClient nClient = new SClient(nSocket);
                nClient.listen();
                Server.clients.add(nClient);

            } catch (IOException ex) {
                System.out.println("There is an error occured when the new client being accepted.");
            }
        }
    }
}

File: Chess/src/main/kotlin/hummel/side_server/ListenConnectionRequestThread.kt
package main.kotlin.hummel.side_server

import java.io.IOException

class ListenConnectionRequestThread(private var server: Server) : Thread() {
	override fun run() {
		while (!server.socket!!.isClosed) {
			try {
				val nSocket = server.socket!!.accept()
				val nClient = SClient(nSocket)
				nClient.listen()
				Server.clients!!.add(nClient)
			} catch (ex: IOException) {
				println("There is an error occurred when the new client being accepted.")
			}
		}
	}
}